x/mongo/driver/topology/pool.go

```diff
@@ -209,7 +209,6 @@ func newPool(config poolConfig, connOpts ...ConnectionOption) *pool {
 		monitor:               config.PoolMonitor,
 		logger:                config.Logger,
 		handshakeErrFn:        config.handshakeErrFn,
-		connOpts:              connOpts,
```

Comment:
What is the benefit of removing the struct field in favor of a method parameter?

Reply:
It is for easy access to the `loadBalanced` field while avoiding calling `newConnectionConfig` repeatedly.

Reply:
How does this change avoid calling `newConnectionConfig` any less than the design as-is? The call to "createConnections" on L238 will call `newConnectionConfig` for every connection.

Reply:
👍, I've passed a `loadBalanced` flag instead of getting the status from `connOpts`.